etna/core/mixins.py

```diff
@@ -89,6 +90,44 @@ def to_dict(self):
         params["_target_"] = BaseMixin._get_target_from_class(self)
         return params
 
+    def set_params(self, **params: dict) -> "BaseMixin":
+        """Return copy of this estimator with given parameters set.
+
+        The method works on simple estimators as well as on nested objects
+        (such as :class:`~etna.pipeline.Pipeline`). The latter have
+        parameters of the form ``<component>.<parameter>`` so that it's
+        possible to update each component of a nested object.
+
+        Parameters
+        ----------
+        **params: dict
+            Estimator parameters.
+
+        """
+        estimator_out = deepcopy(self)
```

Comment:
I think the main problem with this solution is that we use already existing object and modify its parameters.

Imagine if this object was fitted with its parameters and then we change them to new values. In this case the inner state is inconsistent with new parameters and it can behave strangely. Probably, it is better to recreate the object instead of modifying.

Suggested way is described in task:
> We suppose to make conversion via to_dict methods (dict -> patching dict -> creation new object from dict)". 

Have you found some problems in this way of solving?

Reply:
There is a  problem.
I've devised the following way of creating new object from dict:

`target_class_name = catboost_model.to_dict()["_target_"].split(".")[-1]`
`target_class = globals()[target_class_name]`
`estimator_out = target_class()`
instead of
`estimator_out = deepcopy(self)`

The problem is, to create an estimator object of class T with this code, class T must be imported in `etna/core/mixins.py`, meaning it becomes obligatory to import any new created estimator in `etna/core/mixins.py`. I believe that this makes the described solution unscalable. 

I've not yet found a solution avoiding this problem, and I will be grateful for any help in finding it.

Reply:
I think we should use `hydra_slayer.get_from_params` function for dynamic creation of objects. We are already using it in our library.